ngramoffset.go

```diff
@@ -357,8 +360,13 @@ func (a *asciiNgramOffset) SizeBytes() int {
 	return 4*len(a.entries) + 4*len(a.chunkOffsets)
 }
 
+type ngramIndexGetStats struct {
+	// NgramsAccessed is the number of ngrams accessed to lookup gram.
+	NgramsAccessed int
+}
```

Comment:
@stefanhengl what else may be interesting to capture from a btree lookup?

I was thinking of adding a new api.Stat around this. Also additionally increasing IndexBytesLoaded by NgramsAccessed * sizeof(ngram)

Reply:
> I don't think `NgramsAccessed * sizeof(ngram)` measures IndexBytesLoaded, does it?

Yes I don't think it accurately reflects the actual IO work done. But its like in other places we decode stuff from the mmap layer. We only count the amount of bytes actually touched, even though that doesn't correspond to actual work done. It is more of an estimate.

What I am hoping for is some metric go up a _lot_ for the guardrails like searches. Then I want to fix it and see that metric go down (as well as time spent).

> Wouldn't `bucketSize * #ngrams we looked up` be a better estimate? However, this would be super imprecise, too, because it doesn't account for cache, page boundaries, or whatever mmap does.

Do we read the full bucket in? If so then probably worth tracking bytes read as its own metric.